src/mudlet-lua/lua/GUIUtils.lua

```diff
@@ -1473,7 +1473,37 @@ function prefix(what, func, fgc, bgc, window)
   resetFormat(window)
 end
 
--- version of replace function that allows for color, by way of cinsertText
+--- Moves the cursor in the given window up a specified number of lines
+--- @param windowName Optional name of the window to use the function on
+--- @param lines Number of lines to move cursor
+--- @param keep_horizontal Optional boolean to specify if horizontal position should be retained
+function moveCursorUp(window, lines, keep_horizontal)
+  if type(window) ~= "string" then lines, window, keep_horizontal = window, "main", lines end
+  lines = tonumber(lines) or 1
+  if not type(keep_horizontal) == "boolean" then keep_horizontal = false end
+  local curLine = getLineNumber(window)
+  if not curLine then error("moveCursorUp: window does not exist", 2) end
```

Comment:
This should be `nil, msg` rather than a runtime error

Reply:
So, return(nil, "window does not exist")... okay.